src/runners/sac.py

```diff
@@ -157,40 +178,41 @@ def eval(self):
 
             # Quickly dump recently-completed episode's experience to the multithread queue,
             # as long as the episode resulted in "success"
-            if self.agent.cfg[
+            if self.exp_config[
                 "record_experience"
             ]:  # and self.metadata['info']['success']:
                 self.file_logger.log("writing experience")
                 self.agent.save_queue.put(experience)
 
-        self.checkpoint_model(ep_ret, self.cfg["max_ep_len"])
+            self.checkpoint_model(ep_ret, j)
+
         self.agent.update_best_pct_complete(info)
 
         return val_ep_rets
 
     ## Sid needs to take care of the actor_critic objects
     def checkpoint_model(self, ep_ret, n_eps):
         if ep_ret > self.best_ret:  # and ep_ret > 100):
-            path_name = f"{self.cfg['model_save_path']}/best_{self.cfg['experiment_name']}_episode_{n_eps}.statedict"
+            path_name = f"{self.runner_config['model_save_dir']}/best_{self.exp_config['experiment_name']}_episode_{n_eps}.statedict"
             self.file_logger.log(
                 f"New best episode reward of {round(ep_ret, 1)}! Saving: {path_name}"
             )
             self.best_ret = ep_ret
             torch.save(self.actor_critic.state_dict(), path_name)
-            path_name = f"{self.cfg['model_save_path']}/best_{self.cfg['experiment_name']}_episode_{n_eps}.statedict"
+            path_name = f"{self.runner_config['model_save_dir']}/best_{self.exp_config['experiment_name']}_episode_{n_eps}.statedict"
             try:
                 # Try to save Safety Actor-Critic, if present
                 torch.save(self.safety_actor_critic.state_dict(), path_name)
             except:
                 pass
 
-        elif self.cfg["save_freq"] > 0 and (n_eps + 1 % self.cfg["save_freq"] == 0):
-            path_name = f"{self.cfg['model_save_path']}/{self.cfg['experiment_name']}_episode_{n_eps}.statedict"
+        elif self.runner_config["save_every_nth_episode"] > 0 and (n_eps + 1 % self.runner_config["save_every_nth_episode"] == 0):
+            path_name = f"{self.runner_config['model_save_dir']}/{self.exp_config['experiment_name']}_episode_{n_eps}.statedict"
```

Comment:
Should be `{model_save_dir}/{experiment_name}/{experiment_name}_{episode}_{n_eps}.statedict`